Rename DestinationLabel on BottomNavigationBar

lukef opened the pull request:
- Renamed `DestinationLabel` to `BottomNavigationBarItem`
- Renamed `BottomNavigationBarItem.labels` to `BottomNavigationBarItem.items`
- Adjusted comments

lukef committed:
rename DestinationLabel and labels properties to BottomNavigationBarItem and items

Hixie commented on packages/flutter/lib/src/material/bottom_navigation_bar.dart:
```diff
@@ -57,91 +57,91 @@ class DestinationLabel {
-/// labels laid out on top of a piece of material. It provies quick navigation
-/// between top-level views of an app and is typically used on mobile. For
-/// larger screens, side navigation may be a better fit.
+/// The bottom navigation bar consists of multiple item in the form of
```
grammar

Hixie commented on packages/flutter/lib/src/material/bottom_navigation_bar.dart:
```diff
@@ -57,91 +57,91 @@ class DestinationLabel {
-/// between top-level views of an app and is typically used on mobile. For
-/// larger screens, side navigation may be a better fit.
+/// The bottom navigation bar consists of multiple item in the form of
+/// labels and/or icons laid out on top of a piece of material. It provides
```
prefer "labels, icons, or both"

lukef committed:
update/fix comments

lukef commented:
I adjusted the comments based on your feedback. I removed the "primarily used on mobile" as it seems to have been pulled from the material spec directly and its the only real use case right now. I left in a note about larger format devices/screens (it's also in the spec) as it could be useful design advice. Let me know if you want that pulled.

Hixie commented on packages/flutter/lib/src/material/bottom_navigation_bar.dart:
```diff
@@ -57,91 +57,91 @@ class DestinationLabel {
-/// larger screens, side navigation may be a better fit.
+/// The bottom navigation bar consists of multiple items in the form of
+/// labels, icons, or both laid out on top of a piece of material. It provides
+/// quick navigation between the top-level views of an app. Note: for larger screens,
```
Please avoid the word "Note".

See https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-empty-prose

Hixie commented on packages/flutter/lib/src/material/bottom_navigation_bar.dart:
```diff
@@ -57,91 +57,91 @@ class DestinationLabel {
-/// between top-level views of an app and is typically used on mobile. For
-/// larger screens, side navigation may be a better fit.
+/// The bottom navigation bar consists of multiple items in the form of
+/// labels, icons, or both laid out on top of a piece of material. It provides
```
grammar nit: i'd put a comma after "both" to make it clear that the next part doesn't only apply to the "both" option.

Hixie commented:
![LGTM](http://junkyard.damowmow.com/563)

lukef commented:
Did you want me to change the text/grammar? I'll be doing another commit later for a few other things so I could incorporate them later.

Hixie commented:
Yes for the changes, but it's fine if you do them in the coming days instead of right away.

lukef committed:
grammatical changes

lukef commented:
Done

Hixie commented:
+1 thanks!

sethladd commented:
Thanks @lukef !

@Hixie is your LGTM sufficient to merge this?

sethladd commented:
@lukef would you like to add yourself to our AUTHORS file? https://github.com/flutter/flutter/blob/master/AUTHORS

lukef committed:
add myself to AUTHORS

lukef commented:
Added myself. Thanks! 👍

Hixie commented:
Merged. FYI @abarth for the API change. We should make sure we've not left any DestinationLabels anywhere.

Luke, would you mind e-mailing flutter-dev to let them know about the API change? Thanks!

Hixie commented:
And thanks for the contribution! Much appreciated!
